src/interface/plugin.js

```diff
@@ -0,0 +1,80 @@
+/* eslint no-unused-vars: */
+/**
+ * Extends uiveri5 functionality
+ * @constructor
+ * @param {Config} config
+ * @param {Object} instanceConfig
+ * @param {Logger} logger
+ */
+function Plugin(config,instanceConfig,logger){
+}
+
+/**
+ * Called after uiveri5 is started but before the webdriver connection is opened.
+ * 
+ * @return {Promise<void>|void} Can return a promise that will be waited to 
+ * be resolved before uiveri5 continues.
+ */
+Plugin.prototype.setup = function() {
+};
+
+/**
+ * Called after webdriver connection is established but before tests are started.
+ * 
+ * @return {Promise<void>|void} Can return a promise that will be waited to 
+ * be resolved before uiveri5 continues.
+ */
+Plugin.prototype.onPrepare = function() {
+};
+
+/**
+ * Called before the suite is started
+ * 
+ * @param {String} suite.name - suite name
+ */
+Plugin.prototype.suiteStarted = function(suite) {
```

Comment:
for jasmine hooks, what about just setting up jasmine reporters ?

Reply:
This is exactly how this method will be called. But I would like to keep the plugin api clean and focused so would prefer to encapsulate the jasmine wrapper.